### PR TITLE
Draft-to-Planned Approval using Workflows

### PR DESCRIPTION
**Pull request by stevtek**
## What changed

* new backend (/workflow-approve) to approve and update BLIs changing from Draft to Planned
* /approve backend changed to /workflow-submit
* Notifications added for submission and approval
* Frontend wired up to use the workflow backends and to use workflow package entities to indicate which were not sent for approval
* UNDER_REVIEW has been removed from the list of BudgetLineItemStatus so it doesn't show in the tag list of counts, but we still need to remove this deprecated status elsewhere
* Fix: add display_name to BaseModel.to_dict
* Fix: set contract_type as Enum

TODOs (follow up):
* more e2e tests
* submitter notes is currently stored in the package and then not displayed anywhere.  We need to display it, but also, maybe it belongs on the workflow or step instance.

## Issue

#967
#1828 
#1829 

## How to test

* create an agreement with BLIs
* submit one or more BLIs to review for Draft-to-Planned changes
* go the Agreements list and then the For Approval tab
* follow the link to Approve (checkmark) the Agreements BLIs
* complete the form and Approve

## Screenshots

_If relevant, e.g. for a front-end feature_

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [x] Form validations updated

## Links

_If relevant, e.g. for a link to a piece of markdown documentation_

### CODE REVIEW

**Review by tdonaworth — APPROVED** [review]

File: frontend/src/api/opsAPI.js [hunk 1]
```diff
@@ -208,12 +212,25 @@ export const opsApi = createApi({
         }),
         addApprovalRequest: builder.mutation({
             query: (body) => ({
-                url: `/approve/`,
+                url: `/workflow-submit/`,
```
Comment (fpigeonjr) [inline frontend/src/api/opsAPI.js:215]: good name choice

File: frontend/src/components/BudgetLineItems/ChangeIcons/ChangeIcons.js [hunk 2]
```diff
@@ -47,6 +47,13 @@ const ChangeIcons = ({
         );
     }
 
+    // let activeWorkflowCurrentStepId;
```
Comment (fpigeonjr) [inline frontend/src/components/BudgetLineItems/ChangeIcons/ChangeIcons.js:50]: commented code here

File: frontend/src/pages/agreements/approve/ApproveAgreement.js [hunk 3]
```diff
@@ -14,8 +14,42 @@ import { getTotalByCans } from "../review/ReviewAgreement.helpers";
...
+        refetchOnMountOrArgChange: true
+    });
+    if (isLoading) {
+        return <h1>Loading...</h1>;
+    }
+    if (error) {
+        console.log(error);
+        return <h1>Oops, an error occurred</h1>;
+    }
+    console.log(data);
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/approve/ApproveAgreement.js:36]: not sure if we need this console log

File: frontend/src/pages/agreements/approve/ApproveAgreement.js [hunk 4]
```diff
@@ -76,16 +116,44 @@ const ApproveAgreement = () => {
...
         setShowModal(true);
         setModalProps({
             heading:
                 "Are you sure you want to approve these budget lines for Planned Status? This will subtract the amounts from the FY budget.",
             actionButtonText: "Approve",
             secondaryButtonText: "Cancel",
-            handleConfirm: () => {
-                alert("Not implemented yet");
-                navigate("/agreements");
+            handleConfirm: async () => {
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/approve/ApproveAgreement.js:154]: 🤩

File: frontend/src/pages/agreements/approve/ApproveAgreement.js [hunk 5]
```diff
@@ -76,16 +116,44 @@ const ApproveAgreement = () => {
...
+                await approveStep();
+                await navigate("/agreements");
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/approve/ApproveAgreement.js:156]: TypeScript says we can remove the `await` on 156

**Review by fpigeonjr — APPROVED** [review]
🎆 Looks good Steve. Made some non-blocking suggestions. Cool to see the workflow working.

File: frontend/src/pages/agreements/approve/ApproveAgreement.js (hunk 3 above)
Comment (stevtek) [reply frontend/src/pages/agreements/approve/ApproveAgreement.js]: removed

File: frontend/src/components/BudgetLineItems/ChangeIcons/ChangeIcons.js (hunk 2 above)
Comment (stevtek) [reply frontend/src/components/BudgetLineItems/ChangeIcons/ChangeIcons.js]: removed

File: frontend/src/pages/agreements/approve/ApproveAgreement.js (hunk 5 above)
Comment (stevtek) [reply frontend/src/pages/agreements/approve/ApproveAgreement.js:155]: hmmm, that's how it's done on ApproveAgreementEditForm's handleDraft

**Pull request merged**